### ISSUE TITLE
Man pages docset?

### ISSUE THREAD

**Issue by thorstenkampe** [zealdocs/zeal#115]
Can Zeal use local man pages (from Cygwin) as a docset source (as described in http://blog.kapeli.com/linux-man-pages-in-dash/)?

**Comment by warmwaffles** [comment]
This would be a nice feature to have

**Comment by abhisekp** [comment]
:+1: Since I can use many of the linux commandline tools w/o cygwin, by install **util-linux-ng**, so it would be a nice feature to have.

**Comment by benklop** [comment]
KDE's manpage kioslave might be a decent place to start with this: https://quickgit.kde.org/?p=kio-extras.git&a=tree&h=e57f85c7dd358d4141bf203c52640229aee15edc&hb=3d8409f93830cb9bf658bd0648b7b1db3dd5c684&f=man

**Comment by Yanpas** [comment]
There are a lot if sites like this http://linux.die.net/man/ which have all man pages online. It will be easier to use them.

**Comment by ssokolow** [comment]
@Yanpas Point. It probably *would* be easier to point HTTrack at one of those and then slap a docset definition on the result for when I don't have network connectivity.

**Comment by innermous** [comment]
Actually, I've been waiting for this :smiling_imp:

**Comment by Yanpas** [comment]
@ssokolow I've acrually meant exporting man pages online to docset format :smile:

**Comment by ssokolow** [comment]
@Yanpas I have no problem doing that myself... I just consider it unacceptably hacky and restrictive for the official "mark as solved" solution.

**Comment by louy2** [comment]
@Kapeli
How is searching manpages implemented in Dash? Does it hardcode manpage location on OS X? Or does it delegate man page search to `apropos` and parse the result?

**Comment by Kapeli** [comment]
@louy2 both, and a few more.

I have hardcoded paths. I also check: `/private/etc/man.conf`, `/etc/manpaths`, `/etc/manpaths.d`, `/bin/bash -l -c "/usr/bin/man -W"`, `/bin/bash -l -c "/usr/bin/man -k ."`

**Comment by Yanpas** [comment]
OS X docset also contains man pages. It would be fine to have them separately, cause OS X is very huge docset.

**Comment by louy2** [comment]
@Yanpas 
Are you sure? @Kapeli said that on OS X manpage is not stored in docset but rendered on the fly.

**Comment by Yanpas** [comment]
Absolutely. OS x is posix compatible os, so there exist all posix C calls like unlink, read, write, fcntl. The problem that search results show a lot of unnecessary functions with same names from foundation framework, Swift and so on

**Comment by Kapeli** [comment]
The OS X docset contains some man pages, yes. This docset is deprecated as Apple switched to a different documentation format in Xcode 8. The new Apple docs can be found here: https://developer.apple.com/reference. It's unlikely they'll update the OS X docset in the future.

**Comment by ilsken** [comment]
Is there any way to have Dash specifically look in a Dash-specific folder for man pages without setting it globally? Does Dash take into account the MANPATH environment variable if it's set in a plist or something?

**Comment by Kapeli** [comment]
Dash takes into account the MANPATH if it was set for `/bin/bash`. The easiest way to add man paths to Dash is probably in `/etc/manpaths`.

Please [contact me](https://kapeli.com/contact) with support requests regarding Dash in the future. This issue tracker is for Zeal.

**Comment by Yanpas** [comment]
High everybody!
I've created script to generate docsets from man, details here https://github.com/Yanpas/mandocset
It uses man2html utility, avaiable on Ubuntu and Debian e.g. In `etc` folder there are two docsets (posix and linux) which are "compiled", you may use them

I wasn't able to find author of this utility. Beware of a bug: this util may go to endless loop with printf.1p file from posix

**Comment by trollixx** [comment]
@Yanpas nice work. You could also add XML feeds to simplify installation of prebuilt docsets.

**Comment by Yanpas** [comment]
may be closed via https://github.com/Kapeli/Dash-User-Contributions/pull/1278

**Comment by amosbird** [comment]
@Yanpas Hi, how can I install your docset into zeal?

**Comment by Yanpas** [comment]
@amosbird High!
1. Download Linux.docset.zip from here `https://github.com/Yanpas/mandocset/tree/master/etc`
2. Etract it to Linux.docset folder
3. Move this folder to Zeal docsets (`C:\Users\%username%\AppData\Local\Zeal\Zeal\docsets` or `~/.local/share/Zeal/Zeal/docsets/`

**Comment by amosbird** [comment]
Thanks!

**Comment by laoshaw** [comment]
I think it is still much more valuable to search manpages on linux on-the-fly, as manpages are installed incrementally over time, the script above only generates static linux and posix manpages which is sub-optimal.

I tried to use the script against /usr/share/man on linux and while zeal recognizes it, click on search result does not produce any output on the right-side pane in zeal, so it's broken.

is it possible to add dynamic-man-page on the fly to zeal?

**Comment by ssokolow** [comment]
Given that Zeal is built using Qt, I'd think the simplest solution would be:

1. Add an option to use KIOSlaves for disk access.
2. Generate an index which uses `man:` URLs.

(If you've got Konqueror installed, try typing `man:ls` in the address bar.)

**Comment by laoshaw** [comment]
that won't work in gnome though, I don't know how Dash does it, the manpages are all gzipped with many symlinks, the best is to locate them based on search terms in zeal, then gunzip and convert them to html on the fly for zeal to display. unfortunately I'm unfamiliar with Qt.

**Comment by ssokolow** [comment]
> that won't work in gnome though

KDE Frameworks 5 *has* been focused on being modular, so don't count KIOSlaves out immediately.

KIOSlaves are the closest Qt equivalent to GIO/GnomeVFS, and one could already argue that Zeal being based on Qt "won't work in gnome" on the same grounds.

**Comment by Yanpas** [comment]
Sounds complicated. Easier would be pipe gzip to pandoc/man2html to pipe in zeal.

**Comment by ssokolow** [comment]
That *is* an option. The `info` KIOSlave's page footer makes it clear that it's based on a customized version of info2html.

**Comment by trollixx** [comment]
Support for docset format plugins is one of the goals of the upcoming refactoring (or rather reimplementation) of the core logic as a separate backend service. Once we have that, plugin(s) should crawl through the locally available man/info/whatever pages, and populate Zeal's search index.

**Comment by laoshaw** [comment]
any progress on this, would be nice to access local manpages via zeal.

**Comment by Quarkay** [comment]
Maybe this will be helpful:

[Use Linux man pages to make docset](https://www.quarkay.com/code/566/use-Linux-man-pages-to-make-docset-for-Zeal-and-Dash-use-man7-org-online-version-actually)

This article was written in Chinese but google translater is really good [Doge]

**Comment by ernstki** [comment]
I'm happy to report that the instructions at @yanpas' [repository](https://github.com/Yanpas/mandocset) work still fine for an Ubuntu 20.04-based system (elementary OS 6.1).

If you'd prefer that the search prefix be `man:`, it appears you can modify the `DocSetPlatformFamily` key in `~/.local/share/Zeal/Zeal/docsets/<WHATEVER_YOU_NAMED_THE>.docset/Contents/Info.plist` after the fact. Just quit and re-open Zeal for the changes to take effect.

![Screenshot of Info.plist for Linux man pages docset, with the 'DocSetPlatformFamily' key highlighted](https://github.com/zealdocs/zeal/assets/4009681/32d1f4f4-6e3c-4929-bb88-8ebfa01bd23b)
